Make the SVG parser interpret `form feed` as whitespace

Ahmad-S792 opened the pull request:
#### 15c6cea827c0761eb8c1b8638fc12568bf4170b3
<pre>
Make the SVG parser interpret `form feed` as whitespace

<a href="https://bugs.webkit.org/show_bug.cgi?id=77755">https://bugs.webkit.org/show_bug.cgi?id=77755</a>
<a href="https://rdar.apple.com/problem/95488677">rdar://problem/95488677</a>

Reviewed by Anne van Kesteren.

This patch is to align WebKit with Chromium / Blink, Gecko / Firefox and Web Specification [1]:

[1] <a href="https://lists.w3.org/Archives/Public/public-svg-wg/2014AprJun/0068.html">https://lists.w3.org/Archives/Public/public-svg-wg/2014AprJun/0068.html</a>

This patch uses &apos;isASCIIWhite&apos; across SVG Parser and code base to enable it to handle &apos;form feed&apos;.

It is inspired by following change in Blink, which enables &apos;leading&apos; and &apos;trailing&apos; whitespace
in SVG Attributes [2]:

[2] <a href="https://src.chromium.org/viewvc/blink?view=revision&amp">https://src.chromium.org/viewvc/blink?view=revision&amp</a>;revision=175785

Credits to Jacob Goldstein  &lt;jacobg@adobe.com&gt; for &apos;test case&apos;.

* Source/WebCore/svg/SVGParserUtilities.cpp:
(parseGlyphName):
* Source/WebCore/svg/SVGParserUtilities.h:
(isSVGSpace): Deleted
(isSVGSpaceOrComma): Updated
(skipOptionalSVGSpaces):
(skipOptionalSVGSpacesOrDelimiter): Both templates
* Source/WebCore/svg/SVGStringList.cpp:
(SVGStringList::parse):
* LayoutTests/svg/transforms/svg-formFeed-as-whitespace.html: Add Test Case
* LayoutTests/svg/transforms/svg-formFeed-as-whitespace-expected.html: Add Test Case Expectation
* LayoutTests/svg/transforms/svg-line-tabulation-as-not-whitespace.html: Add Test Case
* LayoutTests/svg/transforms/svg-line-tabulation-as-not-whitespace-expected.html: Add Test Case Expectation

Canonical link: <a href="https://commits.webkit.org/273353@main">https://commits.webkit.org/273353@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/384b9bff54f16729d033c88715ca7c775465e642

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35187 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14120 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/37313 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/37942 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/31757 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/36349 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/16502 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11187 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/30666 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/35732 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/11945 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31379 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10471 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/10526 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/31485 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39190 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32008 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/31815 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/36514 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/10651 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/8581 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/34524 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12423 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/31139 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8054 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11175 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/11485 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (approved):
Ideally there would be web platform tests coverage for this as well. And maybe include coverage that U+000B is not whitespace.

Overall though this seems like a good change and helps with cleaning up our whitespace handling: https://bugs.webkit.org/show_bug.cgi?id=255467.

Thanks!

annevk reviewed (approved):
Yeah, sorry, I should have been clearer. "Ideal" wasn't meant to be a merge blocker. Thanks for adding the additional coverage.

Ahmad-S792 committed:
Make the SVG parser interpret `form feed` as whitespace

https://bugs.webkit.org/show_bug.cgi?id=77755
rdar://problem/95488677

Reviewed by Anne van Kesteren.

This patch is to align WebKit with Chromium / Blink, Gecko / Firefox and Web Specification [1]:

[1] https://lists.w3.org/Archives/Public/public-svg-wg/2014AprJun/0068.html

This patch uses 'isASCIIWhite' across SVG Parser and code base to enable it to handle 'form feed'.

It is inspired by following change in Blink, which enables 'leading' and 'trailing' whitespace
in SVG Attributes [2]:

[2] https://src.chromium.org/viewvc/blink?view=revision&revision=175785

Credits to Jacob Goldstein  <jacobg@adobe.com> for 'test case'.

* Source/WebCore/svg/SVGParserUtilities.cpp:
(parseGlyphName):
* Source/WebCore/svg/SVGParserUtilities.h:
(isSVGSpace): Deleted
(isSVGSpaceOrComma): Updated
(skipOptionalSVGSpaces):
(skipOptionalSVGSpacesOrDelimiter): Both templates
* Source/WebCore/svg/SVGStringList.cpp:
(SVGStringList::parse):
* LayoutTests/svg/transforms/svg-formFeed-as-whitespace.html: Add Test Case
* LayoutTests/svg/transforms/svg-formFeed-as-whitespace-expected.html: Add Test Case Expectation
* LayoutTests/svg/transforms/svg-line-tabulation-as-not-whitespace.html: Add Test Case
* LayoutTests/svg/transforms/svg-line-tabulation-as-not-whitespace-expected.html: Add Test Case Expectation

Canonical link: https://commits.webkit.org/273353@main
